### PR TITLE
refactoring(cmd/state): remove submit-tx from cli

### LINKED ISSUE #3345: CLI/API: very unclear how to use `state submit-tx`
how is the user supposed to sign the tx in question? even weirder the tx seems to be expected to be [hex encoded](https://github.com/celestiaorg/celestia-node/blob/332739f9e68196333bd1dea76fbeaac0f12737f8/nodebuilder/state/cmd/state.go#L158-L165)? even if one used app or some code to generate and sign the tx, it is unclear why then go through the hassle to then submit it via node.

Bare minimum is that:
- [ ] someone familiar with node / celestia can look at docs and will know exactly how to create a Tx and sign to submit via SubmitTx (all encoding questions resolved + code snippet)

Optional/Nicetohave:
- [ ] clarify in the docs to the spec so someone who wants to do this e.g. in Rust has all info how to do this or add in another end-point for generating and signing (needs further discussion)

Additionally:
- [ ] consider removing submitx from the CLI (this does not seem useful as is right now)

### PR DESCRIPTION
**Pull request by vgonkivs**
Addresses a TODO in #3345

### CODE REVIEW

**Review by distractedm1nd — APPROVED** [review]
Could we have the doc PR lined up before we merge this?

**Comment by ramin** [comment]
image

**Review by ramin — APPROVED** [review]

File: nodebuilder/state/cmd/state.go [hunk 1]
```diff
@@ -142,29 +140,6 @@ var transferCmd = &cobra.Command{
...
-		if err != nil {
-			return fmt.Errorf("failed to decode tx: %w", err)
-		}
-		txResponse, err := client.State.SubmitTx(
-			cmd.Context(),
-			decoded,
-		)
-		return cmdnode.PrintOutput(txResponse, err, nil)
-	},
-}
```
Comment (liamsi) [inline nodebuilder/state/cmd/state.go:166]: Feels good, doesn't it? 😄

**Review by renaynay — APPROVED** [review]
Please mark as `break` @vgonkivs

**Comment by renaynay** [comment]
Oh i already did 😂

**Review by walldiss — APPROVED** [review]

**Comment by jcstein** [comment]
this still shows up when generating the docs on v0.13.5 https://node-rpc-docs.celestia.org/?version=v0.13.5#state.SubmitTx

**Pull request merged**